internal/cephfs/controllerserver.go

```diff
@@ -794,7 +794,37 @@ func (cs *ControllerServer) CreateSnapshot(
 	}
 
 	if parentVolOptions.BackingSnapshot {
-		return nil, status.Error(codes.InvalidArgument, "cannot snapshot a snapshot-backed volume")
+		err = store.AddSnapshotBackedVolumeRef(ctx, parentVolOptions, cs.ClusterName, cs.SetMetadata, req.GetSecrets())
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+
+		cephfsSnap, genSnapErr := store.GenSnapFromOptions(ctx, req)
+		if genSnapErr != nil {
+			return nil, status.Error(codes.Internal, genSnapErr.Error())
+		}
+
+		sid, _, er := store.CheckSnapExists(ctx, parentVolOptions, cephfsSnap, cs.ClusterName, cs.SetMetadata, cr)
```

Comment:
use `err` instead of `er`?

Reply:
Did it because the variable was shadowing the previous declaration of err.